core/lib/components/os/balenaos.js

```diff
@@ -110,29 +110,55 @@ module.exports = class BalenaOS {
 					fs.createWriteStream(this.image.path),
 				);
 
-				const version = /VERSION="(.*)"/g.exec(
-					await imagefs.readFile({
-						image: this.image.path,
-						partition: 1,
-						path: '/os-release',
-					}),
-				);
-				const variant = /VARIANT_ID="(.*)"/g.exec(
-					await imagefs.readFile({
-						image: this.image.path,
-						partition: 1,
-						path: '/os-release',
-					}),
-				);
+				const res = download.releaseInfo || { version: null, variant: null };
+
+				const readOsRelease = async (pattern, field) => {
+					this.logger.log(`Checking ${field} in os-release`);
+					try {
+						const value = pattern.exec(
+							await imagefs.readFile({
+								image: this.image.path,
+								partition: 1,
+								path: '/os-release',
+							}),
+						);
+						if (value) {
+							res[field] = value[1];
+							this.logger.log(
+								`Found ${field} in os-release file: ${res[field]}`,
+							);
+						}
+					} catch (e) {
+						// If os-release file isn't found, look inside the image to be flashed
+						// Especially in case of OS image inside flasher images. Example: Intel-NUC
+						try {
+							const value = pattern.exec(
+								await imagefs.readFile({
+									image: this.image.path,
+									partition: 2,
+									path: '/etc/os-release',
+								}),
+							);
+							if (value) {
+								res[field] = value[1];
+								this.logger.log(
+									`Found ${field} in os-release file: ${res[field]}`,
+								);
+							}
+						} catch (e) {
+							this.logger.log(`Cannot detect ${field} with os-release`);
+						}
+					}
```

Comment:
```suggestion
							}
						} catch (err) {
							this.logger.log(`Cannot detect ${field} with os-release. Error: ${err}`);
						}
					}
```
I think an error report would be most helpful since we would be already one level deep of error inside an error. Did you ever see how it was being represented? Hope it isn't very confusing. Oh if you do like this suggestion, you might need to run prettify again. I am pretty sure eslint would like to spilt the lines again.